File: src/main/java/com/example/the_machine/dto/FileRefDto.java
package com.example.the_machine.dto;

import java.util.UUID;

public record FileRefDto(
    UUID assetId,
    String filename,
    String mime,
    Long size
) implements ContentBlockDto {

  @Override
  public String type() {
    return ContentBlockTypes.FILE_REF;
  }
}

File: src/main/kotlin/com/example/the_machine/dto/content/FileRefDto.kt
package com.example.the_machine.dto.content

import kotlinx.serialization.Contextual
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.*

@Serializable
@SerialName(ContentBlockTypes.FILE_REF)
data class FileRefDto(
  val assetId: @Contextual UUID,
  val filename: String,
  val mime: String,
  val size: Long
) : ContentBlockDto